xchange-vega/src/main/java/org/knowm/xchange/vega/service/VegaMarketDataService.java

```diff
@@ -0,0 +1,117 @@
+package org.knowm.xchange.vega.service;
+
+import io.vegaprotocol.vega.Markets;
+import io.vegaprotocol.vega.Vega;
+import io.vegaprotocol.vega.api.Trading;
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.marketdata.Ticker;
+import org.knowm.xchange.dto.marketdata.Trade;
+import org.knowm.xchange.dto.marketdata.Trades;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.service.marketdata.MarketDataService;
+import org.knowm.xchange.utils.DateUtils;
+import org.knowm.xchange.vega.VegaAdapters;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.*;
+
+public class VegaMarketDataService extends VegaMarketDataServiceRaw implements MarketDataService {
+
+    private Map<CurrencyPair, String> currencyPairs;
+
+    public VegaMarketDataService(Exchange exchange) {
+        super(exchange);
+    }
+
+    public Map<CurrencyPair, String> getCurrencyPairs() {
+        if (currencyPairs == null) {
+            currencyPairs = new HashMap<>();
+
+            for (Markets.Market market : this.getAllMarkets()) {
+                currencyPairs.put(VegaAdapters.adaptMarketToCurrencyPair(market), market.getId());
+            }
+        }
+
+        return currencyPairs;
+    }
+
+    public List<CurrencyPair> getExchangeSymbols() {
+        return new ArrayList<>(getCurrencyPairs().keySet());
+    }
+
+    @Override
+    public Ticker getTicker(CurrencyPair currencyPair, Object... args) throws IOException {
```

Comment:
please try to implement error mapping as described in: https://github.com/knowm/XChange/wiki/New-Implementation-Best-Practices#step-7